packages/color-scale-block/src/components/SquareWithColor.tsx

```diff
@@ -0,0 +1,161 @@
+        .join('')}`; // Puts the array to together to a string
+};
+
+const getRgbaString = (currentColor: ColorProps) => {
+    if (!(currentColor && currentColor.color)) {
+        return '';
+    }
+
+    return rgbaToHex(
+        `rgba(${currentColor.color.red}, ${currentColor.color.green}, ${currentColor.color.blue}, ${currentColor.color.alpha})`
+    ).toUpperCase();
+};
+
+export const SquareWithColor: FC<SquareWithColorProps> = ({
+    id,
+    index,
+    width,
+    height,
+    isFirst,
+    isLast,
+    currentColor,
+    backgroundColorRgba,
+    onResizeStart,
+    calculateLeftPosition,
+    isEditing,
+    deleteColor,
+    listId,
+    setCurrentlyDraggedColorId,
+    isDragging,
+}) => {
+    const onDrag = () => {
+        if (isDragging !== !!currentColor.id) {
+            setCurrentlyDraggedColorId(currentColor.id);
+        }
+    };
+
+    const onDragEnd = () => {
+        setCurrentlyDraggedColorId(null);
+    };
+
+    const [{}, drag] = useDrag({
+        item: currentColor,
+        collect: (monitor: DragSourceMonitor) => {
+            return {
+                componentDragState: monitor.isDragging() ? ItemDragState.Dragging : ItemDragState.Idle,
+            };
+        },
+        type: listId,
+        canDrag: isEditing,
+    });
+
+    const [copied, setCopied] = useState(false);
+
+    const copyToClipboard = async (text?: string) => {
+        try {
+            const toCopy = text || location.href;
+            await navigator.clipboard.writeText(toCopy);
+        } catch (err) {
+            console.error('Failed to copy: ', err);
+        }
+    };
+
+    const onTooltipClick: MouseEventHandler = (event: MouseEvent) => {
+        event.preventDefault();
+
+        copyToClipboard(getRgbaString(currentColor));
+
+        setCopied(true);
+
+        setTimeout(() => {
+            setCopied(false);
+        }, 2000);
+    };
+
+    const bgColor = `rgba(${backgroundColorRgba}`;
+
+    return (
+        <div
+            id={`row-id-${id}-index-${index}`}
+            style={{
+                width: `${width}px`,
+                left: `${calculateLeftPosition(index, width)}px`,
+                height,
+            }}
+            className={`hover:tw-z-30 row tw-overflow-x-visible tw-overflow-y-hidden tw-pb-8 tw-inline-block
+            `}
+        >
+            <DragHandle index={index} currentColor={currentColor} isEditing={isEditing} onResizeStart={onResizeStart} />
+            <div
+                ref={drag}
+                draggable
+                onDrag={onDrag}
+                onDragEnd={onDragEnd}
+                style={{
+                    height,
+                    width: `${width}px`,
+                }}
+                className={
+                    'tw-group tw-overflow-y-hidden tw-overflow-x-visible tw-top-2 tw-border-[0px] tw-border-white tw-w-full'
+                }
+            >
+                <Tooltip
```

Comment:
Try to use the `TooltipContent` I sent to you. You can extend it for your purpose if needed.

Reply:
It seems that you've already taken care of this in your commit 👍